File: android/src/androidTest/java/org/ligi/gobandroid_hd/uitest/TheTsumegoActivity.java
package org.ligi.gobandroid_hd.uitest;

import android.support.test.filters.MediumTest;
import com.squareup.spoon.Spoon;

import org.ligi.gobandroid_hd.App;
import org.ligi.gobandroid_hd.R;
import org.ligi.gobandroid_hd.TestApp;
import org.ligi.gobandroid_hd.etc.AppModule;
import org.ligi.gobandroid_hd.logic.CellImpl;
import org.ligi.gobandroid_hd.model.GameProvider;
import org.ligi.gobandroid_hd.ui.tsumego.TsumegoActivity;
import org.ligi.gobandroid_hd.base.BaseIntegration;

import javax.inject.Inject;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withText;
import static org.hamcrest.CoreMatchers.not;
import static org.ligi.gobandroid_hd.base.GoViewActions.placeStone;

public class TheTsumegoActivity extends BaseIntegration<TsumegoActivity> {

    @Inject
    GameProvider gameProvider;

    public TheTsumegoActivity() {
        super(TsumegoActivity.class);
    }

    @Override
    public void setUp() throws Exception {
        super.setUp();
        TestApp.component().inject(this);
    }

    @MediumTest
    public void testThatNoTsumegoWarningComes() {
        gameProvider.set(readGame("default_marker"));
        final TsumegoActivity activity = getActivity();

        onView(withText(R.string.tsumego_sgf_no_solution)).check(matches(isDisplayed()));
        Spoon.screenshot(activity, "tsumego_fail");
    }

    @MediumTest
    public void testThatOffPathMessageIsNotThereInBeginning() {
        gameProvider.set(readGame("tsumego"));
        final TsumegoActivity activity = getActivity();

        onView(withId(R.id.tsumego_off_path_view)).check(matches(not(isDisplayed())));
        Spoon.screenshot(activity, "tsumego_on_path");
    }

    @MediumTest
    public void testThatOffPathMessageComes() {
        gameProvider.set(readGame("tsumego"));
        final TsumegoActivity activity = getActivity();

        onView(withId(R.id.go_board)).perform(placeStone(new CellImpl(1, 1), gameProvider.get()));

        onView(withId(R.id.tsumego_off_path_view)).check(matches(isDisplayed()));
        Spoon.screenshot(activity, "tsumego_off_path");
    }

    @MediumTest
    public void testThatCommentComesAndGoes() {
        gameProvider.set(readGame("tsumego"));
        final TsumegoActivity activity = getActivity();
        onView(withId(R.id.game_comment)).check(matches(withText("testing comment")));

        Spoon.screenshot(activity, "tsumego_comment");
        onView(withId(R.id.go_board)).perform(placeStone(new CellImpl(1, 1), gameProvider.get()));

        onView(withId(R.id.game_comment)).check(matches(not(isDisplayed())));
        Spoon.screenshot(activity, "tsumego_comment_gone");
    }



}

File: android/src/androidTest/java/org/ligi/gobandroid_hd/uitest/TheTsumegoActivity.kt
package org.ligi.gobandroid_hd.uitest

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.*
import com.github.salomonbrys.kodein.instance
import com.github.salomonbrys.kodein.lazy
import org.hamcrest.CoreMatchers.not
import org.junit.Rule
import org.junit.Test
import org.ligi.gobandroid_hd.App
import org.ligi.gobandroid_hd.R
import org.ligi.gobandroid_hd.logic.CellImpl
import org.ligi.gobandroid_hd.model.GameProvider
import org.ligi.gobandroid_hd.test_helper_functions.placeStone
import org.ligi.gobandroid_hd.test_helper_functions.readGame
import org.ligi.gobandroid_hd.ui.tsumego.TsumegoActivity
import org.ligi.trulesk.TruleskActivityRule

class TheTsumegoActivity {

    @get:Rule
    val rule = TruleskActivityRule(TsumegoActivity::class.java, false)

    val gameProvider: GameProvider by App.kodein.lazy.instance()

    @Test
    fun testThatNoTsumegoWarningComes() {
        gameProvider.set(readGame("default_marker")!!)
        val activity = rule.launchActivity()

        onView(withText(R.string.tsumego_sgf_no_solution)).check(matches(isDisplayed()))
        rule.screenShot("tsumego_fail")
    }

    @Test
    fun testThatOffPathMessageIsNotThereInBeginning() {
        gameProvider.set(readGame("tsumego")!!)
        val activity = rule.launchActivity(null)

        onView(withId(R.id.tsumego_off_path_view)).check(matches(not(isDisplayed())))
        rule.screenShot("tsumego_on_path")
    }

    @Test
    fun testThatOffPathMessageComes() {
        gameProvider.set(readGame("tsumego")!!)
        val activity = rule.launchActivity()

        onView(withId(R.id.go_board)).perform(placeStone(CellImpl(1, 1), gameProvider.get()))

        onView(withId(R.id.tsumego_off_path_view)).check(matches(isDisplayed()))
        rule.screenShot("tsumego_off_path")
    }

    @Test
    fun testThatCommentComesAndGoes() {
        gameProvider.set(readGame("tsumego")!!)
        val activity = rule.launchActivity()
        onView(withId(R.id.game_comment)).check(matches(withText("testing comment")))

        rule.screenShot("tsumego_comment")
        onView(withId(R.id.go_board)).perform(placeStone(CellImpl(1, 1), gameProvider.get()))

        onView(withId(R.id.game_comment)).check(matches(not(isDisplayed())))
        rule.screenShot("tsumego_comment_gone")
    }

}
